graphql-database-manager-dynamo/src/main/java/com/fleetpin/graphql/database/manager/dynamo/DynamoDb.java

```diff
@@ -405,37 +436,43 @@ public int maxBatchSize() {
+        var destination = client.updateItem(request -> request.tableName(entityTable).key(key).conditionExpression("attribute_exists(links)").updateExpression("SET links.#table = :val ADD revision :revisionIncrement").expressionAttributeNames(k).expressionAttributeValues(values).returnValues(ReturnValue.UPDATED_NEW))
                 .handle((r, e) -> {
                     if (e != null) {
                         if (e.getCause() instanceof ConditionalCheckFailedException) {
                             Map<String, AttributeValue> m = new HashMap<>();
                             m.put(target, values.get(":val"));
                             values.put(":val", AttributeValue.builder().m(m).build());
-                            return client.updateItem(request -> request.tableName(entityTable).key(key).conditionExpression("attribute_not_exists(links)").updateExpression("SET links = :val").expressionAttributeValues(values));
+                            values.put(":revisionIncrement", REVISION_INCREMENT);
+                            
+                            return client.updateItem(request -> request.tableName(entityTable).key(key).conditionExpression("attribute_not_exists(links)").updateExpression("SET links = :val ADD revision :revisionIncrement").expressionAttributeValues(values).returnValues(ReturnValue.UPDATED_NEW));
                         } else {
                             throw new RuntimeException(e);
                         }
                     }
-                    return CompletableFuture.completedFuture(null);
+                    return CompletableFuture.completedFuture(r);
                 }).thenCompose(a -> a)
                 .handle((r, e) -> {
                     if (e != null) {
                         if (e.getCause() instanceof ConditionalCheckFailedException) {
-                            return client.updateItem(request -> request.tableName(entityTable).key(key).conditionExpression("attribute_exists(links)").updateExpression("SET links.#table = :val").expressionAttributeNames(k).expressionAttributeValues(values));
+                            return client.updateItem(request -> request.tableName(entityTable).key(key).conditionExpression("attribute_exists(links)").updateExpression("SET links.#table = :val ADD revision :revisionIncrement").expressionAttributeNames(k).expressionAttributeValues(values).returnValues(ReturnValue.UPDATED_NEW));
                         } else {
                             throw new RuntimeException(e);
                         }
                     }
-                    return CompletableFuture.completedFuture(null);
+                    return CompletableFuture.completedFuture(r);
                 }).thenCompose(a -> a);
         future = future.thenCombine(destination, (a, b) -> b);
         return future.thenApply(response -> {
+        	entity.setRevision(Integer.parseInt(response.attributes().get("revision").n()));
             return entity;
         });
     }
```

Comment:
This method is now a behemoth in size...
I'm finding it a little hard to follow the whole thing.
Could we cut it up into smaller pieces with some aptly named methods?

Reply:
It only selected one line, but it's for the whole "link" method.

Reply:
done